applovin_max/ios/Classes/AppLovinMAXAdView.m

```diff
@@ -5,30 +5,88 @@
+static NSMutableDictionary<NSString *, AppLovinMAXAdViewPlatformWidget *> *preloadedPlatformWidgetInstances;
 
 + (void)initialize
 {
     [super initialize];
-    adViewInstances = [NSMutableDictionary dictionaryWithCapacity: 2];
+    platformWidgetInstances = [NSMutableDictionary dictionaryWithCapacity: 2];
+    preloadedPlatformWidgetInstances = [NSMutableDictionary dictionaryWithCapacity: 2];
 }
 
 + (MAAdView *)sharedWithAdUnitIdentifier:(NSString *)adUnitIdentifier
 {
-    return adViewInstances[adUnitIdentifier];
+    return (preloadedPlatformWidgetInstances[adUnitIdentifier] ?: platformWidgetInstances[adUnitIdentifier]).adView;
 }
 
+#pragma mark - Preloading
+
++ (void)preloadPlatformWidgetAdView:(NSString *)adUnitIdentifier 
+                           adFormat:(MAAdFormat *)adFormat
+                          placement:(nullable NSString *)placement  
+                         customData:(nullable NSString *)customData
+                    extraParameters:(nullable NSDictionary<NSString *, id> *)extraParameters
+               localExtraParameters:(nullable NSDictionary<NSString *, id> *)localExtraParameters
+                         withResult:(FlutterResult)result
+{
+    AppLovinMAXAdViewPlatformWidget *preloadedPlatformWidget = preloadedPlatformWidgetInstances[adUnitIdentifier];
+    if ( preloadedPlatformWidget )
+    {
+        result([FlutterError errorWithCode: @"AppLovinMAX" message: @"Cannot preload more than one for a single Ad Unit ID." details: nil]);
+        return;
+    }
+    
+    preloadedPlatformWidget = [[AppLovinMAXAdViewPlatformWidget alloc] initWithAdUnitIdentifierForPreload: adUnitIdentifier adFormat: adFormat preload: YES];
+    preloadedPlatformWidgetInstances[adUnitIdentifier] = preloadedPlatformWidget;
+    
+    preloadedPlatformWidget.placement = placement;
+    preloadedPlatformWidget.customData = customData;
+    preloadedPlatformWidget.extraParameters = extraParameters;
+    preloadedPlatformWidget.localExtraParameters = localExtraParameters;
+    
+    [preloadedPlatformWidget loadAd];
+    
+    result(nil);
+}
+
++ (void)destroyPlatformWidgetAdView:(NSString *)adUnitIdentifier withResult:(FlutterResult)result
+{
+    AppLovinMAXAdViewPlatformWidget *preloadedPlatformWidget = preloadedPlatformWidgetInstances[adUnitIdentifier];
+    if ( !preloadedPlatformWidget )
+    {
+        result([FlutterError errorWithCode: @"AppLovinMAX" message: @"No native UI component found to destroy" details: nil]);
+        return;
+    }
+    
+    if ( [preloadedPlatformWidget hasContainerView] )
+    {
+        result([FlutterError errorWithCode: @"AppLovinMAX" message: @"Cannot destroy - currently in use" details: nil]);
+        return;
+    }
+    
+    [preloadedPlatformWidgetInstances removeObjectForKey: adUnitIdentifier];
+    
+    [preloadedPlatformWidget detachAdView];
+    [preloadedPlatformWidget destroy];
+    
+    result(nil);
+}
+
+#pragma mark - AdView
+
 - (instancetype)initWithFrame:(CGRect)frame
```

Comment:
nit since this is initializer, this should be declared before

```
+ (void)preloadPlatformWidgetAdView:(NSString *)adUnitIdentifier 
                           adFormat:(MAAdFormat *)adFormat
                          placement:(nullable NSString *)placement  
                         customData:(nullable NSString *)customData
                    extraParameters:(nullable NSDictionary<NSString *, id> *)extraParameters
               localExtraParameters:(nullable NSDictionary<NSString *, id> *)localExtraParameters
                         withResult:(FlutterResult)result
```

Reply:
And pragma mark should be `#pragma mark - Initializer`.

Reply:
Static functions should come first ?, so I added the pragma mark for `+ (void)initialize`.

Reply:
Not necessarily - in our codebases, we declare static initializers first, then initializers, then anything else in between 🤠